crates/floresta-chain/src/pruned_utreexo/mod.rs

```diff
@@ -351,23 +354,58 @@ impl<T: BlockchainInterface> BlockchainInterface for Arc<T> {
     ) -> Result<Stump, Self::Error> {
         T::update_acc(self, acc, block, height, proof, del_hashes)
     }
-
     fn get_chain_tips(&self) -> Result<Vec<BlockHash>, Self::Error> {
         T::get_chain_tips(self)
     }
-
     fn validate_block(
         &self,
         block: &Block,
         proof: Proof,
         inputs: HashMap<OutPoint, TxOut>,
         del_hashes: Vec<sha256::Hash>,
-        acc: Stump,
+        chain_time: ChainTime,
     ) -> Result<(), Self::Error> {
-        T::validate_block(self, block, proof, inputs, del_hashes, acc)
+        T::validate_block(self, block, proof, inputs, del_hashes, chain_time)
     }
 
     fn get_fork_point(&self, block: BlockHash) -> Result<BlockHash, Self::Error> {
         T::get_fork_point(self, block)
     }
 }
+
+pub mod chain_time {
+    use core::time::Duration;
+    /// Here we delegate to the consumer of the library to implement the [`ChainTimeInterface`] so we dont need to get stuck with std::time.
+    ///
+    /// Take a look on how we implement the [`ChainTimeInterface`] for the [`ChainTime`] struct if you want to implement your own.
+    pub trait ChainTimeInterface {
+        /// This should return a zero time for test cases. While validating, if we receive a zero time, Time-Related checks should be skipped.
+        fn zero() -> ChainTime;
+        /// This should return a unix timestamp.
+        fn get_time(&self) -> Duration;
+        /// This should return a new instance of ChainTime.
+        fn new(time: Duration) -> ChainTime;
+    }
+    /// We need a some local time notation while validating blocks.
+    ///
+    /// floresta-chain relies on the [`ChainTime`] to have some block validation.
+    #[derive(Debug, Clone, Copy)]
+    pub struct ChainTime(Duration);
```

Comment:
Shouldn't be `Instant`? Why using `Duration` here?

Reply:
Can be

Reply:
Actually cannot, it belongs to std